### PR TITLE
Create proper overloads for fragment binding.

### LINKED ISSUE #95: "bindFragment" Method Signature
It takes `Object` and uses ugly `instanceof` and runtime API checks. Why not just have two overloads?

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #95. Refs #136.

### CODE REVIEW

**Comment by JakeWharton** [comment]
Proof running:

**Comment by mttkay** [comment]
:+1:

**Comment by hamidp** [comment]
Nice, :+1:

**Comment by JakeWharton** [comment]
Thanks.

File: rxandroid/src/main/java/rx/android/app/AppObservable.java [hunk 1]
```diff
@@ -93,17 +80,20 @@ public Boolean call(android.support.v4.app.Fragment fragment) {
...
+        return source.observeOn(mainThread())
+                .lift(new OperatorConditionalBinding<T, Fragment>(fragment, FRAGMENT_VALIDATOR));
+    }
+
+    /**
+     * Binds the given source sequence to a support-v4 fragment.
+     *
+     * @see #bindFragment(Fragment, Observable)
+     */
+    public static <T> Observable<T> bindSupportFragment(android.support.v4.app.Fragment fragment, Observable<T> source) {
```
Comment (rosshambrick) [inline rxandroid/src/main/java/rx/android/app/AppObservable.java]: Is there something that prevents using the same method name?
`public static <T> Observable<T> bindFragment(android.support.v4.app.Fragment fragment, Observable<T> source)`

**Pull request merged**